gamedata.py

```diff
@@ -720,8 +721,28 @@ def _get_cards(self) -> Generator[tuple[str, dict[str, str]], None, None]:
             except KeyError:
                 meta = {"CHARACTER": "Special"}
             yield card, meta
+    
+    @property
+    @abstractmethod
+    def removals(self) -> list[str]:
+        pass
+
+    def _get_removals(self) -> Generator[tuple[str, dict[str, str]], None, None]:
+        for card in self.removals:
```

Comment:
this section could maybe be a shared function with line 702 above it, but I wasn't exactly sure how returns work with yields and things like that.

https://github.com/Vgr255/TwitchCordBot/blob/c5552503d0dd3b3dccbbdde1b35a64df253bb94c/gamedata.py#L702

Reply:
I also don't know how they interact. I'm sure this isn't difficult to find out, but no worries, as I'm fine keeping these separate.

Reply:
I tried and it caused some errors. Was hoping you guys would know if I was doing it wrong lol.

```
def _get_cards(self) -> Generator[tuple[str, dict[str, str]], None, None]:
        if "master_deck" in self._data:
            return self._get_card_data(self._data["master_deck"]) <----- method reference here, this is wrong

       ...

@property
abstractmethod
def removals(self) -> list[str]:
    pass

# HERE'S THE SHARED METHOD
def _get_card_data(self, cards) -> Generator[tuple[str, dict[str, str]], None, None]:
    for card in cards:
        try:
            meta = get_card_metadata(card)
        except KeyError:
            meta = {"CHARACTER": "Special"}
        yield get_card(card), meta
    return

def removals_as_html(self) -> Generator[str, None, None]:
    return self._cards_as_html(self._get_card_data(self.removals)) <-------- method reference here, this is fine
```